### PR TITLE
[activation] ELU

### PR DESCRIPTION
**Pull request by KSTconsult**

### CODE REVIEW

File: src/runtime/cpu/kernels/kernels.hpp [hunk 1]
```diff
@@ -89,6 +89,23 @@ void CPUReLUForward(ConstVector X, Vector result);
 */
 void CPUReLUInputGrad(ConstVector input, ConstVector grad_output, Vector grad_input);
 
+/*
+ * CPUELUForward
+ * implements computation of ELU layer output
+ * X is a 1D vector [size]
+ * result have size [size]
+*/
+void CPUELUForward(ConstVector X, Vector result, float alpha);
```
Comment (l-bat) [inline src/runtime/cpu/kernels/kernels.hpp:98]: Do we need to add documentation for `alpha`?

**Review by l-bat — COMMENTED** [review]
Looks great!

File: tests/unit/kernels_reference.cpp [hunk 2]
```diff
@@ -104,6 +104,20 @@ void dw::reference::CPUReLUBackward(const float* in, const float* grad_output,
                    [](float in, float go) { return in > 0.0 ? go : 0.0;});
 }
 
+void CPUELUForward(const float* in, float* out, float alpha, size_t size) {
```
Comment (TolyaTalamanov) [inline tests/unit/kernels_reference.cpp:107]: Stop using raw pointers in reference signature, because it isn't descriptive

**Review by TolyaTalamanov — COMMENTED** [review]
Looks good, but is there any test for that ?

File: src/runtime/cpu/kernels/kernels.hpp (hunk 1 above)
Comment (KSTconsult) [reply src/runtime/cpu/kernels/kernels.hpp:105]: Done.

File: tests/unit/kernels_reference.cpp (hunk 2 above)
Comment (KSTconsult) [reply tests/unit/kernels_reference.cpp]: Done.

File: tests/unit/kernels_reference.cpp [hunk 3]
```diff
@@ -104,6 +104,27 @@ void dw::reference::CPUReLUBackward(const float* in, const float* grad_output,
                    [](float in, float go) { return in > 0.0 ? go : 0.0;});
 }
 
+void dw::reference::CPUELUForward(const Tensor& in, Tensor& out, float alpha) {
+    float* raw_in = in.data();
+    float* raw_out = out.data();
+
+    for (size_t i = 0; i < in.total(); i++) {
+        raw_out[i] = raw_in[i] < 0.f ? alpha * (std::exp(raw_in[i]) - 1) : raw_in[i];
```
Comment (TolyaTalamanov) [inline tests/unit/kernels_reference.cpp:112]: Just wondering why it isn't `transofrm` as for `CPUELUBackward` let's keep consistence. Chose the single solution for that

**Review by TolyaTalamanov — APPROVED** [review]
LGTM  👍

**Review by l-bat — APPROVED** [review]

File: tests/unit/test_cpuelu.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#include "kernels_reference.hpp"
+#include "test_utils.hpp"
+
+namespace dw = deepworks;
+
+TEST(LayerTests, CPUELUForward) {
+    int in_features = 100;
+    int batch_size  = 32;
+    dw::Placeholder in(dw::Shape{batch_size, in_features});
+    dw::Model model(in, dw::ELU(0.2 /* alpha */, "elu")(in));
```
[the remark this answers is not in the document]
Comment (KSTconsult) [reply tests/unit/test_cpuelu.cpp]: Done.

File: tests/unit/kernels_reference.cpp (hunk 3 above)
Comment (KSTconsult) [reply tests/unit/kernels_reference.cpp]: Done.

**Review by TolyaTalamanov — APPROVED** [review]
Great!
LTGM :+1:

**Pull request merged**